compute_tools/src/http/api.rs

```diff
@@ -98,6 +99,38 @@ async fn routes(req: Request<Body>, compute: &Arc<ComputeNode>) -> Response<Body
             }
         }
 
+        (&Method::POST, "/extensions/install") => {
```

Comment:
I think a `/extensions` endpoint would be better, but fairly subjective critique.

```
GET /extensions # list of all extensions. right now we are using /installed_extensions, but we can change that in the future
DELETE /extensions # uninstall an extension
POST /extensions # install an extension
```

Reply:
> Disagree with /extensions as name, as that implies plural, while this endpoint only handles a singular extension.

That is typical REST API design. See our public API for creating a project: https://api-docs.neon.tech/reference/createproject.

Reply:
It also wouldn't be inconceivable to just take an array for the endpoint anyway.

Reply:
To me the most REST way would be just `POST` to `/extensions`, it implies we're adding/installing one. But also this is not a public API so it doesn't matter as much. I'm fine with either.

Reply:
I've been wanting compute_ctl to grow a more REST-like interface anyway. This is a good start. I'd eventually like to take a dependency on something like axum to structure this code better.

Reply:
> IMO, doing this at the top level of /extensions doesn't make much sense - we can't list the installed extensions without parameterizing on database or making the endpoint slow by O(n databases). A not-quite-REST endpoint under /extensions/install makes more sense to me.

Isn't this was Anastasia's work was all about? We just move here code into the `GET` side of the endpoint.

Reply:
> Isn't this was Anastasia's work was all about? We just move her code into the GET side of the endpoint.

Yeah, this https://github.com/neondatabase/neon/pull/8917 added an `/installed_extensions` API

Don't really have a strong opinion. Converging to
```
POST /extensions
```
would work for me

Leaving `/installed_extensions` as also probably make sense because just 'extensions' is vague, and installed extensions != available

Reply:
> Leaving /installed_extensions as also probably make sense because just 'extensions' is vague, and installed extensions != available

```
GET /extensions?installed=true
```

was my feedback in the original PR.